pkg/k8s/pipelineRun.go

```diff
@@ -360,7 +379,7 @@ func (r *pipelineRun) changeStatusAndUpdateSafely(change func() error) error {
 			defer func() { isRetry = true }()
 		}
 
-		changeError = change()
+		changeError = r.applyChangeWithoutUpdate(change)
```

Comment:
This here does not work in case of a retry. In case of a re-try the in-memory representation is dropped and replaced by a fresh copy from storage. Would be required to apply the changes also on that copy. But we do not know which changes has been applied.